From 50b5b8cb013c89b04d450539d3272356d379b2dc Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 19 Feb 2023 19:37:20 +0200
Subject: [PATCH] [SW-99614] gaudi2: security: refactor rrs usage

Refactor static definitions of RRs for both LBW and HBW to enable better
garbage collection during FW components compilation (linkage).
To make this possible split the static arrays to single per type
definitions and update functions using the arrays to use the new
definitions.
This allows garbage collector to remove the dead code from Gaudi2 ppboot
and eROM PCIe. The removal from each component saves approx. 260 bytes,
otherwise occupied by dead code.

Change-Id: Ia0cfc1bf5dc46d025eaef5f1a7e0a79eccbaa58a
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 46ce12e..cb9a96c 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -59,89 +59,90 @@
 #include <gaudi2_fw_if.h>
 #include "common.h"
 
-static struct hl_sec_rrs_cfg hl_sec_lbw_rrs_tbl[] = {
-	[HL_SEC_ENGINE_IF_RTR] = {
-		.instances_dist = HL_SEC_IF_RTR_DIST,
-		.min_short_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
-		.max_short_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
-		.min_full_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
-		.max_full_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
-		.hit_wr_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
-	},
-	[HL_SEC_ENGINE_PCIE] = {
-		.instances_dist = HL_SEC_PCIE_INST_DIST,
-		.min_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
-		.max_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
-		.min_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
-		.max_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
-		.hit_wr_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
-	},
-	[HL_SEC_ENGINE_SFT] = {
-		.instances_dist = 0,
-		.min_short_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
-		.max_short_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
-		.min_full_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
-		.max_full_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
-		.hit_wr_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
-	}
+static struct hl_sec_rrs_cfg hl_sec_lbw_if_rtr_rrs = {
+	.instances_dist = HL_SEC_IF_RTR_DIST,
+	.min_short_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
+	.max_short_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
+	.min_full_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
+	.max_full_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
+	.hit_wr_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
 };
 
-static struct hl_sec_rrs_cfg_hbw hl_sec_hbw_rrs_tbl[] = {
-	[HL_SEC_ENGINE_IF_RTR] = {
-		.instances_dist = HL_SEC_IF_RTR_DIST,
-		.min_short.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
-		.min_short.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
-		.max_short.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
-		.max_short.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
-		.min_full.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
-		.min_full.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
-		.max_full.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
-		.max_full.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
-		.hit_wr_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
-	},
-	[HL_SEC_ENGINE_PCIE] = {
-		.instances_dist = 0,
-		.min_short.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
-		.min_short.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
-		.max_short.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
-		.max_short.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
-		.min_full.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
-		.min_full.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
-		.max_full.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
-		.max_full.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
-		.hit_wr_breg = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
-	},
-	[HL_SEC_ENGINE_PSOC] = {
-		.instances_dist = HL_SEC_PSOC_INST_DIST,
-		.min_short.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
-		.min_short.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
-		.max_short.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
-		.max_short.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
-		.min_full.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
-		.min_full.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
-		.max_full.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
-		.max_full.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
-		.hit_wr_breg = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
-	},
-	[HL_SEC_ENGINE_SFT] = {
-		.instances_dist = HL_SEC_HBW_SFT_DIST,
-		.min_short.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
-		.min_short.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
-		.max_short.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
-		.max_short.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
-		.min_full.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
-		.min_full.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
-		.max_full.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
-		.max_full.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
-		.hit_wr_breg = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
-		.hit_rd_breg = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
-	},
+static struct hl_sec_rrs_cfg hl_sec_lbw_pcie_rrs = {
+	.instances_dist = HL_SEC_PCIE_INST_DIST,
+	.min_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
+	.max_short_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
+	.min_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
+	.max_full_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
+	.hit_wr_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
+};
+
+static struct hl_sec_rrs_cfg hl_sec_lbw_sft_rrs = {
+	.instances_dist = 0,
+	.min_short_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_SHORT_0,
+	.max_short_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_SHORT_0,
+	.min_full_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0,
+	.max_full_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0,
+	.hit_wr_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmSFT0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_HIT_AR,
+};
+
+static struct hl_sec_rrs_cfg_hbw hl_sec_hbw_if_rtr_rrs = {
+	.instances_dist = HL_SEC_IF_RTR_DIST,
+	.min_short.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
+	.min_short.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
+	.max_short.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
+	.max_short.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
+	.min_full.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
+	.min_full.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
+	.max_full.lo = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
+	.max_full.hi = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
+	.hit_wr_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmDCORE0_RTR0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
+};
+
+static struct hl_sec_rrs_cfg_hbw hl_sec_hbw_pcie_rrs = {
+	.instances_dist = 0,
+	.min_short.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
+	.min_short.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
+	.max_short.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
+	.max_short.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
+	.min_full.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
+	.min_full.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
+	.max_full.lo = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
+	.max_full.hi = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
+	.hit_wr_breg = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmPCIE_MSTR_RR_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
+};
+
+static struct hl_sec_rrs_cfg_hbw hl_sec_hbw_psoc_rrs = {
+	.instances_dist = HL_SEC_PSOC_INST_DIST,
+	.min_short.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
+	.min_short.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
+	.max_short.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
+	.max_short.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
+	.min_full.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
+	.min_full.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
+	.max_full.lo = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
+	.max_full.hi = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
+	.hit_wr_breg = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
+};
+
+static struct hl_sec_rrs_cfg_hbw hl_sec_hbw_sft_rrs = {
+	.instances_dist = HL_SEC_HBW_SFT_DIST,
+	.min_short.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_LO_0,
+	.min_short.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_SHORT_HI_0,
+	.max_short.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_LO_0,
+	.max_short.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_SHORT_HI_0,
+	.min_full.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0,
+	.min_full.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0,
+	.max_full.lo = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0,
+	.max_full.hi = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0,
+	.hit_wr_breg = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AW,
+	.hit_rd_breg = mmSFT0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_HIT_AR,
 };
 
 static struct hl_sec_cfg_hmmu hmmu_cfg = {
@@ -235,12 +236,11 @@
 	hl_secure_update_hit_reg(rr_idx, inst_offset, &rr_reg->hit, write_only);
 }
 
-static void hl_secure_hbw_rr_gen(int rr_idx, u64 min, u64 max,
-		int dist, int inst_num, enum hl_sec_engine_type type)
+static void hl_secure_hbw_rr_gen(int rr_idx, u64 min, u64 max, int dist,
+				 int inst_num, struct hl_sec_rrs_cfg_hbw *cfg)
 {
 	int i;
 	struct hbw_rr_reg rr_reg;
-	struct hl_sec_rrs_cfg_hbw *cfg = &hl_sec_hbw_rrs_tbl[type];
 
 	rr_reg.data_min = min >> HL_SEC_SHIFT_4K_POS;
 	rr_reg.data_max = max >> HL_SEC_SHIFT_4K_POS;
@@ -281,10 +281,9 @@
 
 /* hl_secure_lbw_rr_gen() function deals only with short RRs rr_idx < 14*/
 static void hl_secure_lbw_rr_gen(int rr_idx, u64 min, u64 max, int dist,
-		int inst_num, enum hl_sec_engine_type type)
+				 int inst_num, struct hl_sec_rrs_cfg *cfg)
 {
 	int i;
-	struct hl_sec_rrs_cfg *cfg = &hl_sec_lbw_rrs_tbl[type];
 	struct lbw_rr_reg rr_reg;
 
 	rr_reg.data_min = (min & ~CFG_BASE);
@@ -312,20 +311,22 @@
 	if (rr_sel.noc) {
 		for (i = 0; i < HL_SEC_NUM_OF_DCORES; i++) {
 			hl_secure_lbw_rr_gen(rr_idx, min, max, HL_SEC_DCORE_DIST * i,
-					NUM_OF_RTR_PER_DCORE, HL_SEC_ENGINE_IF_RTR);
+					     NUM_OF_RTR_PER_DCORE,
+					     &hl_sec_lbw_if_rtr_rrs);
 
 			hl_secure_lbw_rr_gen(rr_idx, min, max, HL_SEC_SFT_DIST * i,
-					HL_SEC_SFT_INST_NUM, HL_SEC_ENGINE_SFT);
+					     HL_SEC_SFT_INST_NUM,
+					     &hl_sec_lbw_sft_rrs);
 		}
 	}
 
 	if (rr_sel.pcie)
 		hl_secure_lbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_INST_NUM,
-					HL_SEC_ENGINE_PCIE);
+				     &hl_sec_lbw_pcie_rrs);
 
 	if (rr_sel.psoc)
 		hl_secure_hbw_rr_gen(HL_SEC_HBW_PSOC_LBW, min, max, 0, HL_SEC_PSOC_INST_NUM,
-					HL_SEC_ENGINE_PSOC);
+				     &hl_sec_hbw_psoc_rrs);
 }
 
 static const struct hl_sec_efuse *preboot_get_hl_sec_fake_efuse(void)
@@ -862,8 +863,8 @@
 	max = CFG_BASE + CFG_SIZE - 1;
 
 	/* Block only pcie RR in ppboot */
-	hl_secure_lbw_rr_gen(HL_SEC_LBW_DBG_CFG_SPACE, min, max,
-				0, HL_SEC_PCIE_INST_NUM, HL_SEC_ENGINE_PCIE);
+	hl_secure_lbw_rr_gen(HL_SEC_LBW_DBG_CFG_SPACE, min, max, 0,
+			     HL_SEC_PCIE_INST_NUM, &hl_sec_lbw_pcie_rrs);
 }
 
 static void preboot_secure_lbw(void)
@@ -936,7 +937,7 @@
 	 * registers alone is not enough to block/allow pcie traffic.
 	 * The dcore0 rtr0 is therefore coupled with pcie rr configuration.
 	 */
-	hl_secure_hbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_HBW_NUM, HL_SEC_ENGINE_PCIE);
+	hl_secure_hbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_HBW_NUM, &hl_sec_hbw_pcie_rrs);
 
 	/*
 	 * In case noc is already enabled along with pcie, then dcore0 rtr0 is configured
@@ -944,7 +945,7 @@
 	 */
 	if (if_rtr)
 		hl_secure_hbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PCIE_RTR_NUM,
-							HL_SEC_ENGINE_IF_RTR);
+				     &hl_sec_hbw_if_rtr_rrs);
 
 }
 
@@ -955,10 +956,11 @@
 	if (rr_sel.noc) {
 		for (i = 0; i < HL_SEC_NUM_OF_DCORES; i++) {
 			hl_secure_hbw_rr_gen(rr_idx, min, max, HL_SEC_DCORE_DIST * i,
-					HL_SEC_NUM_OF_RTR_PER_DCORE, HL_SEC_ENGINE_IF_RTR);
+					     HL_SEC_NUM_OF_RTR_PER_DCORE,
+					     &hl_sec_hbw_if_rtr_rrs);
 
 			hl_secure_hbw_rr_gen(rr_idx, min, max, HL_SEC_SFT_DIST * i,
-					HL_SEC_HBW_SFT_INST_NUM, HL_SEC_ENGINE_SFT);
+					     HL_SEC_HBW_SFT_INST_NUM, &hl_sec_hbw_sft_rrs);
 		}
 	}
 
@@ -967,7 +969,8 @@
 		hl_secure_hbw_pcie_rr(rr_idx, min, max, !rr_sel.noc);
 
 	if (rr_sel.psoc)
-		hl_secure_hbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PSOC_INST_NUM, HL_SEC_ENGINE_PSOC);
+		hl_secure_hbw_rr_gen(rr_idx, min, max, 0, HL_SEC_PSOC_INST_NUM,
+				     &hl_sec_hbw_psoc_rrs);
 }
 
 #define DBG_SPACE_BASE		0x1000007FF8000000ull
